app/src/main/java/com/framgia/music_51/data/TrackDataSource.java

```diff
@@ -0,0 +1,12 @@
+package com.framgia.music_51.data;
+
+import com.framgia.music_51.data.model.GenreType;
+import com.framgia.music_51.data.model.MusicResponse;
+
+import io.reactivex.Single;
+
+public interface TrackDataSource {
+    interface Remote {
+        Single<MusicResponse> getTracksOnGenre(String kind, @GenreType String type);
```

Comment:
```suggestion
        Single<MusicResponse> getTracks(String kind, @GenreType String type);
```
nên sử dụng overload.